test/unit/exceptions/test_exception_handler.py

```diff
@@ -1,127 +0,0 @@
-    callback_called = True
-    stage = stage_param
-    reason = reason_param
-
-
-def test_exception_handler_with_function_callback() -> None:
-    @exception_handler("test_1", callback=callback)
-    def fut() -> None:
-        raise Exception("failure")
-
-    with pytest.raises(SystemExit):
-        fut()
-
-    assert callback_called
-    assert stage == "test_1"
-    assert str(reason) == str(Exception("failure"))
-
-
-def test_exception_handler_with_method_callback() -> None:
-    callback_instance = CallbackClass()
-
-    @exception_handler("test_2", callback=callback_instance.callback)
-    def fut() -> None:
-        raise Exception("failure_2")
-
-    with pytest.raises(SystemExit):
-        fut()
-
-    assert callback_instance.callback_called
-    assert callback_instance.stage == "test_2"
-    assert str(callback_instance.reason) == str(Exception("failure_2"))
-
-
-def test_exception_handler_with_system_exit() -> None:
-    callback_instance = CallbackClass()
-
-    @exception_handler("test_2", callback=callback_instance.callback)
-    def fut() -> None:
-        sys.exit(11)
-
-    with pytest.raises(SystemExit):
-        fut()
-
-    assert callback_instance.callback_called
-    assert callback_instance.stage == "test_2"
-    assert str(callback_instance.reason) == str(Exception("System exit with code:11"))
-
-
-def test_exception_handler_decorates_method() -> None:
-    callback_instance = CallbackClass()
-
-    class Fut:
-        @exception_handler("test", callback=callback_instance.callback)
-        def fut(self) -> None:
-            raise Exception("failure")
-
-    with pytest.raises(SystemExit):
-        Fut().fut()
-
-    assert callback_instance.callback_called
-    assert callback_instance.stage == "test"
-    assert str(callback_instance.reason) == str(Exception("failure"))
-
-
-def test_exception_memory_error_thrown() -> None:
-    callback_instance = CallbackClass()
-    exc_message = "Out of available memory"
-
-    @exception_handler("test_3", callback=callback_instance.callback)
-    def fut() -> None:
-        raise MemoryError(exc_message)
-
-    with pytest.raises(SystemExit):
-        fut()
-
-    assert callback_instance.callback_called
-    assert callback_instance.stage == "test_3"
-    assert str(callback_instance.reason) == str(
-        RuntimeMemoryException(str(MemoryError(exc_message)))
-    )
-
-
-def test_exception_missing_dependency_error_thrown() -> None:
-    callback_instance = CallbackClass()
-    exc_message_pattern = re.compile(".*No module named 'missing_dep'.*")
-
-    @exception_handler("test_4", callback=callback_instance.callback)
-    def fut() -> None:
-        import importlib
-
-        importlib.import_module("missing_dep")
-        return
-
-    with pytest.raises(SystemExit):
-        fut()
-
-    assert callback_instance.callback_called
-    assert callback_instance.stage == "test_4"
-    assert exc_message_pattern.match(str(callback_instance.reason))
```

Comment:
Why do we only test these for the model decorator now, but not for dataset?

Reply:
We're going to merge more and more of models and datasets together (in a subsequent PR). For now, testing these for models should be sufficient as datasets go through almost the exact same codepath.